cibuildwheel/linux.py

```diff
@@ -8,29 +8,19 @@
 
 from .docker_container import DockerContainer
 from .logger import log
-from .util import (BuildOptions, BuildSelector, NonPlatformWheelError,
-                   get_build_verbosity_extra_flags, prepare_command)
-
-
-def matches_platform(identifier: str) -> bool:
-    pm = platform.machine()
-    if pm == "x86_64":
-        # x86_64 machines can run i686 docker containers
-        if identifier.endswith('x86_64') or identifier.endswith('i686'):
-            return True
-    elif pm == "i686":
-        if identifier.endswith('i686'):
-            return True
-    elif pm == "aarch64":
-        if identifier.endswith('aarch64'):
-            return True
-    elif pm == "ppc64le":
-        if identifier.endswith('ppc64le'):
-            return True
-    elif pm == "s390x":
-        if identifier.endswith('s390x'):
-            return True
-    return False
+from .util import (
+    Architecture, BuildOptions, BuildSelector, NonPlatformWheelError,
+    get_build_verbosity_extra_flags, prepare_command,
+)
+
+
+re_pattern = re.compile(r'[cp]p\d{2}-manylinux_(\w*)')
+
+
+def matches_platform(identifier: str, architectures: List[Architecture]) -> bool:
+    matched_architecture = re_pattern.search(identifier)
+    id_architecture = matched_architecture.group(1) if matched_architecture else ''
+    return id_architecture in architectures
```

Comment:
I had a comment on this earlier, which seems to have been lost - do people think the following might be simpler?

```suggestion
    return any(identifier.endswith(arch) for arch in supported_platforms)```

Reply:
The only reason I went with a regex pattern was because I wasn't sure we will always match the architecture at the end of each identifier. Though thats probably premature optimization if we don't expect that to change any time soon.